### PR TITLE
Multiple enhancements for WCT 4.0

### PR DESCRIPTION
**Pull request by jeromelaban**
- Adds support for `IconElement` fast conversion
- Adds stubs for `ToggleSwitchTemplateSettings`, `PackageId`, `UISettings`
- Adjust `XamlObjectBuilder` logging
- Add implicit conversion for `KeyTime` and `Duration`
- Add support for top level `StaticResource` resource dictionary elements
- Implement FindFirstParent for net46/netstd2.0
- Adds ElementNotAvailableException and ElementNotEnabledException
- Fix invalid measure for empty wasm images
- Add size/rect checks for measure/arrange wasm
-  Improve XamlReader error reporting

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Shapes/ArbitraryShapeBase.Android.cs [hunk 1]
```diff
@@ -116,8 +116,10 @@ private IDisposable BuildDrawableLayer()
             var imageBrushFill = Fill as ImageBrush;
             if (imageBrushFill != null)
             {
-					var bitmapDrawable = new BitmapDrawable(imageBrushFill.TryGetBitmap(drawArea, () => RefreshShape(forceRefresh: true), path));
-					drawables.Add(bitmapDrawable);
+#pragma warning disable CS0618 // Type or member is obsolete
```
Comment (Dissolute) [inline src/Uno.UI/UI/Xaml/Shapes/ArbitraryShapeBase.Android.cs]: It's fine to hide warnings in legacy code, but we shouldn't be doing it for current code... there's no way to address the warning?

File: src/Uno.UWP/Generated/3.0.0.0/Windows.UI.ViewManagement/UISettings.cs [hunk 2]
```diff
@@ -1,241 +1,163 @@
...
 namespace Windows.UI.ViewManagement
 {
-	#if __ANDROID__ || __IOS__ || NET46 || false
-	[global::Uno.NotImplemented]
-	#endif
 	public  partial class UISettings 
 	{
-		#if __ANDROID__ || __IOS__ || NET46 || false
-		[global::Uno.NotImplemented]
 		public  bool AnimationsEnabled
```
Comment (Dissolute) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.UI.ViewManagement/UISettings.cs]: Not marked [NotImplemented]

File: src/Uno.UWP/Generated/3.0.0.0/Windows.UI.ViewManagement/UISettings.cs [hunk 3]
```diff
@@ -1,241 +1,163 @@
...
 		{
 			get
 			{
-				throw new global::System.NotImplementedException("The member bool UISettings.AnimationsEnabled is not implemented in Uno.");
+				global::Windows.Foundation.Metadata.ApiInformation.TryRaiseNotImplemented("Windows.UI.ViewManagement.UISettings", "AnimationsEnabled");
+#if __WASM__
```
Comment (Dissolute) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.UI.ViewManagement/UISettings.cs]: ??

Comment (Dissolute) [reply src/Uno.UWP/Generated/3.0.0.0/Windows.UI.ViewManagement/UISettings.cs]: I guess WCT checks this even though we don't? Comment needed here

File: src/Uno.UWP/Generated/3.0.0.0/Windows.UI.ViewManagement/UISettings.cs [hunk 4]
```diff
@@ -1,241 +1,163 @@
...
+#else
+				return true;
+#endif
 			}
 		}
-		#endif
-		#if __ANDROID__ || __IOS__ || NET46 || __WASM__
-		[global::Uno.NotImplemented]
+
 		public  uint CaretBlinkRate
```
Comment (Dissolute) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.UI.ViewManagement/UISettings.cs]: These should all still be [NotImplemented]

File: src/Uno.UI/UI/Xaml/Controls/Primitives/ToggleSwitchTemplateSettings.cs [hunk 5]
```diff
@@ -0,0 +1,80 @@
+#pragma warning disable 108 // new keyword hiding
+#pragma warning disable 114 // new keyword hiding
+namespace Windows.UI.Xaml.Controls.Primitives
+{
+	public  partial class ToggleSwitchTemplateSettings : global::Windows.UI.Xaml.DependencyObject
```
[the remark this answers is not in the document]
Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/Primitives/ToggleSwitchTemplateSettings.cs]: It's used in templates, it was raising quite a few exceptions but for now we'll just return default values instead to be less noisy.

**Pull request merged**